segment/segment.go

```diff
@@ -0,0 +1,77 @@
+package segment
+
+import "encoding/hex"
+
+var (
+	separator = []byte{'\n'}
+	pool      = hashPool{}
+)
+
+// Marshal returns a segment for the given identity and value.
+func Marshal(v SelfAccessor, identity Identity) (Segment, error) {
+	return MarshalWith(Wrap(v), v, identity)
+}
+
+// MarshalWith returns a segment for the given identity and value, using the specified accessor.
+func MarshalWith(accessor FieldAccessor, v interface{}, identity Identity) (Segment, error) {
+	h := pool.GetHash()
+	seg := NewSegment(identity)
```

Comment:
It seems inefficient to use an object here, where a string would do.

Reply:
The thing is, a string will not do. I want the segment to contain the data used to identify it, as well as the original values for the reverse lookup. There have been use-cases where this was needed.